### PR TITLE
Test for TeacherServiceImpl - addTeacherRoleToUser

### LINKED ISSUE #108: Test for TeacherServiceImpl - addTeacherRoleToUser
Unit test for the **addTeacherRoleToUser** method:

- [ ] Verify that it correctly adds the teacher role to a user when the user doesn't already have that role.
- [ ] Verify that it throws a UserIsAlreadyATeacherException when the user already has the teacher role.

_**instructions for the task**_
_use the convention //given //when //then
mock all injected dependencies with business logic using the Mockito library, e.g. UserRepository
try to use the BDD methodology, e.g. the assertJ library (optional)

### PR DESCRIPTION
**Pull request by GrzegorzHolui**

### CODE REVIEW

File: src/test/java/pl/simpleascoding/tutoringplatform/user/TeacherServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    private static final String NAME = "TestName";
+
+    private static final String SURNAME = "TestSurname";
+
+    private static final String EMAIL = "account@server.com";
+
+    @InjectMocks
+    private TeacherServiceImpl teacherService;
+    @Mock
+    private UserService userService;
```
Comment (dawciobiel) [inline src/test/java/pl/simpleascoding/tutoringplatform/user/TeacherServiceImplTest.java:41]:
```suggestion
    @InjectMocks
    private TeacherServiceImpl teacherService;
    
    @Mock
    private UserService userService;
```

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Please fix the code formatting.

File: src/test/java/pl/simpleascoding/tutoringplatform/user/TeacherServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        user.setRoles(Set.of(RoleType.TEACHER));
+        SignAsTeacherDTO signAsTeacherDTO = new SignAsTeacherDTO(USERNAME);
+        given(userService.getUserByUsername(USERNAME)).willReturn(user);
+
+        //when
+        Throwable thrown = catchThrowable(() -> teacherService.addTeacherRoleToUser(signAsTeacherDTO));
+
+        //then
+        verify(userService, times(1)).getUserByUsername(USERNAME);
+        AssertionsForClassTypes.assertThat(thrown)
```
Comment (wh15tl3r) [inline src/test/java/pl/simpleascoding/tutoringplatform/user/TeacherServiceImplTest.java]: Unnecessary `AssertionsForClassTypes`; use `assertThat` directly.

**Review by wh15tl3r — CHANGES_REQUESTED** [review]

**Review by dawciobiel — APPROVED** [review]
LGTM

**Review by wh15tl3r — APPROVED** [review]
LGTM

**Pull request merged**